### ISSUE TITLE
The nanite guide to create a super soldier(aka cockroach) via bug exploit

### ISSUE THREAD

**Issue by davethwave** [tgstation/tgstation#42681]
Round ID:101393

Reproduction:You go and give someone dermal hardening with the timer on any amount of time(lets say 10 seconds) and you set the timer type to self delete. You activate the nanites and when the time is up and self deletes the person still keeps the extra armor values instead of having them removed. This works with the stun shortening nanites and a few lethal nanites like death sim which will leave the victim in a fake death coma for the rest of the round. This allows the ability to be invincible to everything with the exception of chems, atmos, cold/heat, gibbing, surgery, their own nanites, and cremation.

**Comment by MMMiracles** [comment]
you die in the simulation you die in real life

**Comment by ATH1909** [comment]
Don't stop believing...

**Comment by ATH1909** [comment]
This is really weird, and I can't tell why this happens from looking at the code.

Does this work with things like the dermal hardening program?

**Comment by davethwave** [comment]
I am not sure, I'll test it.

**Comment by davethwave** [comment]
It appears that the bug works with all the programs as far as I can see.

**Comment by davethwave** [comment]
It doesn't seem to work with the infective exo-locomotion nanites or any regen nanites however. I believe the problem is that while the nanites currently work with things like the exo locomotion the stuff that effect states like death, or armor don't get changed back to their earlier state for some reason when they stop working.

**Comment by ATH1909** [comment]
Can you stack the effects of dermal hardening in this manner?

**Comment by davethwave** [comment]
I'm not sure. I'll test it tomorrow

**Comment by davethwave** [comment]
A easy way to test this though is to get a nanite remote, set the activation code to whatever, set the timer to 1 and the type to deactivate. Inject a ssd, humanoid monkey, or something similar and spam click them with the remote. Then use a saw on them or something with high brute damage. If the thing didn't dent the guy then we have a very powerful exploit on our hands. Also make sure that your using dermal hardening for this specific test.

**Comment by HideAndSeekLOGIC** [comment]
Reason for this is that cloud backup automatically activates/deactivates any programs which are in such a state on the cloud, as well as restoring any deleted programs.

**Comment by ATH1909** [comment]
> Reason for this is that cloud backup automatically activates/deactivates any programs which are in such a state on the cloud, as well as restoring any deleted programs.

But this bug works even if your nanites aren't cloud-synced.

**Comment by davethwave** [comment]
Did some more testing and here are the results, if you set the timer type to self-delete and put that on a nanite server, the people will within 2 or so minutes be impossible to kill via normal means anymore. Each hit of burn and brute will always deal only one damage and stuns will do nothing against this super human army you created. The only way to kill them is via cremation, gibbing, chems, suffocation, heat/cold, or surgery. However if you set it for the disable instead of self delete it only stacks 2 times at most.

**Comment by ATH1909** [comment]
> Did some more testing and here are the results, if you set the timer type to self-delete and put that on a nanite server, the people will within 2 or so minutes be impossible to kill via normal means anymore. Each hit of burn and brute will always deal only one damage and stuns will do nothing against this super human army you created. The only way to kill them is via cremation, gibbing, chems, suffocation, heat/cold, or surgery. However if you set it for the disable instead of self delete it only stacks 2 times at most.

https://www.youtube.com/watch?v=SwveuIRWxBo

I think this is worthy of being set to high priority, right?

(Also, you can make those burn and brute hits deal 0 damage by getting fully augged with robotic parts, as they give you damage reduction (or was it damage deflection? I forget which) against brute and burn damage. Also, you're not gonna be immune to non-[melee, bullet, laser, or energy] types of damage, although that definitely doesn't make this NOT absurdly overpowered (It does mean that you can stop/damage these monsters with elances, though).)

**Comment by davethwave** [comment]
don't even need the 0 damage, just put accelerated regen on the nanites and they are are impossible to kill via brute or burn. Also I tested this at round end, and getting shot several dozen times with lasers and bullets and getting mauled by a ripley didn't even do anything to me. The only thing that really stopped me was the disarm that drops me for a tenth of a second making me drop my weapon.

**Comment by ATH1909** [comment]
> don't even need the 0 damage, just put accelerated regen on the nanites and they are are impossible to kill via brute or burn. Also I tested this at round end, and getting shot several dozen times with lasers and bullets and getting mauled by a ripley didn't even do anything to me. The only thing that really stopped me was the disarm that drops me for a tenth of a second making me drop my weapon.

Yeah, but the 0-ing of brute and burn damage from those attack types rubs it in even further (especially if you combine the augs with the mechanical repair program, which can continuously heal you without causing you to have a negative net nanite regeneration rate (albeit at a slightly slower rate than the accelerated regeneration program)). I guess being augged would make you vulnerable to emps, though...

Do the effects persist if the cloud ID of your uber-nanites gets deleted or if you lose all of your nanites (as in, your nanite level drops to 0 and you'll need to be reinjected (or have admins intervene or somethign) to have nanites in your body again)?

**Comment by ATH1909** [comment]
Do they have to be active when they're deleted?

If so, you might want to wait a bit before beginning the uber nanite-ification process, as the programs might shut off once your nanite levels drop down to the safety threshold.

By my calculations, if you want your uber-nanites to apply the effects of the dermal hardening, dermal refractive support, and nerve support programs once every 30 seconds, you should build up a nanite reserve of the safety threshold (50 by default, IIRC)+10 nanite units per intended cycle.

If you want to throw in the rapid coagulation program in there as well you should build up a nanite reserve of the safety threshold+11 nanite units per intended cycle.

If you want to throw in the mental barrier and electrical conduction programs (in addition to the rapid coagulation program) as well (they're two passive effects that won't self-stack, but should still be able to be permanently applied for free with this method), you should build up a nanite reserve of the safety threshold+17 nanite units per intended cycle.

**Comment by davethwave** [comment]
Eh, what I did was just use a remote and had the nanites set up a timer to self delete on 10 seconds rather than set up the nanite server so that it would automatically do that. The cost of a 10 second usage is insignificant usually so I just waited a few minutes till I had 180 then while I was on the shuttle activated the nantites occasionally. I activated it about 4-5 times so that would be about 10 nanites per buff due to nanite regen speed, and since I was using a remote there was a few seconds extra of time to build up the buffer. So you don't need a very large buffer to do this at all. Honestly the biggest surprise with this is how nobody has paid any attention about this super soldier talk we have here or have made any attempts on fixing it. Though it's probably my fault due to the issue name. Death Sim nanites don't stop don't speak of invincibility or being impossible to kill after all. I'll rename it I guess.
